acvm/src/lib.rs

```diff
@@ -207,6 +207,9 @@ pub trait ProofSystemCompiler {
     /// The Error type returned by failed function calls in the ProofSystemCompiler trait.
     type Error: std::error::Error; // fully-qualified named because thiserror is `use`d at the top of the crate
 
+    /// Returns the identifier for the backend.
+    fn backend_identifier(&self) -> String;
```

Comment:
This is also needed by code that calls the CommonReferenceString trait, so do we want it to be on the Backend trait itself or a separate trait that others can depend upon?

Reply:
Hmm, I don't really want to put this on `Backend` as that would hamper our ability to pass `&impl SubTraits` around effectively.

I placed it on `ProofSystemCompiler` as this is the trait which is most tightly bound to the underlying proving library. `PWG`/`SmartContract` shouldn't ever need to expose this information so we could have a separate trait which is just `ProofSystemCompiler` and `CommonReferenceString` depend on.